lib/mayaHydra/hydraExtensions/mayaHydraLibInterface.h

```diff
@@ -17,15 +17,16 @@
 #ifndef MAYAHYDRALIB_INTERFACE_H
 #define MAYAHYDRALIB_INTERFACE_H
 
-#include <mayaHydraLib/api.h>
+//Local headers
+#include "mayaHydraLib/api.h"
 
+//Hydra headers
 #include <pxr/imaging/hd/sceneIndex.h>
 
-#include <vector>
 
 PXR_NAMESPACE_OPEN_SCOPE
 
-using SceneIndicesVector = std::vector<HdSceneIndexBasePtr>;
+using SceneIndicesVector                                        = std::vector<HdSceneIndexBasePtr>;//Be careful, these are not not Ref counted. Elements could become dangling
```

Comment:
I think it's just a comment to be aware than when using SceneIndicesVector it's not ref counted but I don't think it can cause problems. But in one of the past version of this code, I was using some vector with custom class than were ref counted so we could have an ambiguity. I would left the comment as it's not always obvious from the code to differentiate between HdSceneIndexBasePtr and HdSceneIndexBaseRefPtr

Reply:
If it's a maintenance burden, then I would change to using HdSceneIndexBaseRefPtr.